src/providers/formattingProvider.ts

```diff
@@ -221,7 +221,7 @@ export class FormatProvider implements vscode.DocumentFormattingEditProvider {
                     let temp: RegExpExecArray;
                     if (
                         !!(temp = new RegExp(
-                            '\\b' + oneCommand + '\\b(.*)',
+                            '^(?:{|}|\\s)*' + oneCommand + '\\b(.*)',
```

Comment:
I'm decent with regular expressions, but this is a big one.

- non-capturing group
- start of line
- open brace or close brace or empty space or nothing
- no longer checking against word breaks at all?

Given the number of changes here, can we add some unit tests to clarify what this regular expression is supposed to (and not supposed to) match?

i.e.

```ts
    test('One command code regex', () => {
        const oneCommand = 'ifnotexist';
        const regex = new RegExp('^(?:{|}|\\s)*' + oneCommand + '\\b(.*)');

        assert.match(oneCommand, regex, 'command on its own');
        assert.doesNotMatch(
            `'${oneCommand}'`,
            regex,
            'command wrapped in quotes',
        );
        // ...more tests here...
    });
```

This would help me merge with confidence :)

Reply:
"before it allowed only curved braces and spaces", can we add tests for those cases as well to ensure we're still allowing them?